### DISCUSSION TITLE
What can be max value for fcst_history?

### DISCUSSION THREAD

**Discussion by RajS999** [timeseriesAI/tsai#720] [Q&A]
Well, the title says it all. But let me elaborate. I was trying out [PatchTST notebook](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/15_PatchTST_a_new_transformer_for_LTSF.ipynb).
It sets `fcst_history` to `104`. Is there any max value for `fcst_history`? and what will be implication of larger values?

**Comment by oguiza** [comment]
Hi @RajS999,
You can use any number of time steps as history. The implication is that Transformers (like PatchTST) require a lot of memory. And it grows quadratically. So if you double the fcst_history you'll require a much larger GPU memory to avoid OOM errors. You may also need to reduce the batch size. In any case, training will be slower, but it might be more accurate.
